File: python/examples/accounts/onlinereturnpolicy/v1/get_online_return_policy_sample.py
# -*- coding: utf-8 -*-
# Copyright 2025 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
"""Retrieves an online return policy using the Merchant API."""

# [START merchantapi_get_online_return_policy]
from examples.authentication import configuration
from examples.authentication import generate_user_credentials
from google.shopping.merchant_accounts_v1 import GetOnlineReturnPolicyRequest
from google.shopping.merchant_accounts_v1 import OnlineReturnPolicyServiceClient

_ACCOUNT = configuration.Configuration().read_merchant_info()


def get_online_return_policy(account_id: str, return_policy_id: str) -> None:
  """Gets an online return policy.

  Args:
    account_id: The merchant account ID.
    return_policy_id: The ID of the return policy to retrieve.
  """

  # Gets OAuth Credentials.
  credentials = generate_user_credentials.main()

  # Creates a client.
  client = OnlineReturnPolicyServiceClient(credentials=credentials)

  # The name has the format:
  # accounts/{account}/onlineReturnPolicies/{return_policy}
  name = f"accounts/{account_id}/onlineReturnPolicies/{return_policy_id}"

  # Creates the request.
  request = GetOnlineReturnPolicyRequest(name=name)

  # Makes the request and catches and prints any error messages.
  try:
    print("Sending Get OnlineReturnPolicy request:")
    response = client.get_online_return_policy(request=request)
    print("Retrieved OnlineReturnPolicy below")
    print(response)
  except RuntimeError as e:
    print(e)


if __name__ == "__main__":
  # The ID of the account from which to get the return policy.
  account_id_ = _ACCOUNT
  # The ID of the return policy to retrieve.
  # This is the `returnPolicyId` field of the `OnlineReturnPolicy` resource.
  return_policy_id_ = "<SET_RETURN_POLICY_ID>"
  get_online_return_policy(account_id_, return_policy_id_)


# [END merchantapi_get_online_return_policy]

File: php/examples/accounts/onlinereturnpolicy/v1/GetOnlineReturnPolicySample.php
<?php
/**
 * Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// [START merchantapi_get_online_return_policy]
require_once __DIR__ . '/../../../../vendor/autoload.php';
require_once __DIR__ . '/../../../Authentication/Authentication.php';
require_once __DIR__ . '/../../../Authentication/Config.php';

use Google\ApiCore\ApiException;
use Google\Shopping\Merchant\Accounts\V1\Client\OnlineReturnPolicyServiceClient;
use Google\Shopping\Merchant\Accounts\V1\GetOnlineReturnPolicyRequest;

/**
 * This class demonstrates how to get an OnlineReturnPolicy for a given
 * Merchant Center account.
 */
class GetOnlineReturnPolicySample
{
    /**
     * A helper function to create the name string.
     *
     * @param string $accountId The merchant account ID.
     * @param string $returnPolicyId The return policy ID.
     *
     * @return string The name has the format:
     *     `accounts/{account}/onlineReturnPolicies/{return_policy}`
     */
    private static function getName(
        string $accountId,
        string $returnPolicyId
    ): string {
        return sprintf(
            'accounts/%s/onlineReturnPolicies/%s',
            $accountId,
            $returnPolicyId
        );
    }

    /**
     * Gets an online return policy from your Merchant Center account.
     *
     * @param array $config The configuration data for authentication and account ID.
     * @param string $returnPolicyId The ID of the return policy to retrieve.
     */
    public static function getOnlineReturnPolicy(
        array $config,
        string $returnPolicyId
    ): void {
        // Gets the OAuth credentials to make the request.
        $credentials = Authentication::useServiceAccountOrTokenFile();

        // Creates options containing credentials for the client to use.
        $options = ['credentials' => $credentials];

        // Creates a client.
        $onlineReturnPolicyServiceClient =
            new OnlineReturnPolicyServiceClient($options);

        // Creates the name of the return policy to retrieve.
        $name = self::getName($config['accountId'], $returnPolicyId);

        // Calls the API and catches and prints any network failures/errors.
        try {
            // Creates the request object.
            $request = new GetOnlineReturnPolicyRequest(['name' => $name]);

            printf("Sending Get OnlineReturnPolicy request:%s", PHP_EOL);
            $response = $onlineReturnPolicyServiceClient->getOnlineReturnPolicy(
                $request
            );

            printf("Retrieved OnlineReturnPolicy below%s", PHP_EOL);
            // The `serializeToJsonString()` method is chosen here for better
            // readability of the output.
            printf('%s%s', $response->serializeToJsonString(true), PHP_EOL);
        } catch (ApiException $e) {
            printf('%s%s', $e->getMessage(), PHP_EOL);
        }
    }

    /**
     * Helper to execute the sample.
     */
    public function callSample(): void
    {
        $config = Config::generateConfig();
        // Replace with the actual return policy ID you want to retrieve.
        $returnPolicyId = '<SET_RETURN_POLICY_ID>';
        self::getOnlineReturnPolicy($config, $returnPolicyId);
    }
}

// Runs the script.
$sample = new GetOnlineReturnPolicySample();
$sample->callSample();
// [END merchantapi_get_online_return_policy]
